internal/service/timestreamquery/scheduled_query.go

```diff
@@ -0,0 +1,955 @@
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/id"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/enum"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource("aws_timestreamquery_scheduled_query", name="Scheduled Query")
+// @Tags(identifierAttribute="arn")
+func newResourceScheduledQuery(_ context.Context) (resource.ResourceWithConfigure, error) {
+	r := &resourceScheduledQuery{}
+
+	r.SetDefaultCreateTimeout(30 * time.Minute)
+	r.SetDefaultUpdateTimeout(30 * time.Minute)
+	r.SetDefaultDeleteTimeout(30 * time.Minute)
+
+	return r, nil
+}
+
+const (
+	ResNameScheduledQuery         = "Scheduled Query"
+	ScheduledQueryFieldNamePrefix = "ScheduledQuery"
+)
+
+type resourceScheduledQuery struct {
+	framework.ResourceWithConfigure
+	framework.WithTimeouts
+}
+
+func (r *resourceScheduledQuery) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_timestreamquery_scheduled_query"
+}
+
+func (r *resourceScheduledQuery) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			names.AttrARN: framework.ARNAttributeComputedOnly(),
+			names.AttrCreationTime: schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+			names.AttrExecutionRoleARN: schema.StringAttribute{
+				Required: true,
+			},
+			names.AttrKMSKeyID: schema.StringAttribute{
+				Optional: true,
+			},
+			names.AttrName: schema.StringAttribute{
+				Required: true,
+			},
+			"next_invocation_time": schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+			"previous_invocation_time": schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+			"query_string": schema.StringAttribute{
+				Required: true,
+			},
+			names.AttrState: schema.StringAttribute{
+				CustomType: fwtypes.StringEnumType[awstypes.ScheduledQueryState](), // ENABLED, DISABLED
+				Computed:   true,
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+		},
+		Blocks: map[string]schema.Block{
+			"error_report_configuration": schema.SingleNestedBlock{
```

Comment:
@YakDriver I think that `SingleNestedBlock` has caused problems in the past and are discouraged. see :https://github.com/hashicorp/terraform-provider-aws/issues/35813